spec/models/Restaurant_spec.rb

```diff
@@ -0,0 +1,32 @@
+require 'rails_helper'
+require 'faker'
+
+describe Restaurant do
+
+  it { should validate_presence_of(:name) }
+  it { should have_many(:reviews) }
+  it { should belong_to(:cuisine) }
+
+  it { should allow_values('SomeString').for(:address) }
+  it { should_not allow_values(Faker::Lovecraft.fhtagn(5)).for(:address) }
+
+  it { should allow_values('35').for(:delivery_time) }
+  it { should_not allow_values('5000').for(:delivery_time) }
+
+  context 'When has multiple reviews' do
+    ratings = [1,3,2,1]   #ratings for restaurant's reviews
+    let!(:restaurant_w_reviews) { FactoryBot.create :restaurant_with_reviews, ratings: ratings}
+    it 'Returns reviews average ratings' do
+      expect(restaurant_w_reviews.rating).to eql(ratings.sum.fdiv(ratings.size))
+    end
+  end
+
+  context 'When has no reviews' do
+    let!(:restaurant) { FactoryBot.create :restaurant }
+    it 'Returns nil for ratings' do
+      expect(restaurant.rating).to be_nil
+    end
+  end
+
```

Comment:
can you also test the associations in the model? I think this is a good practice